source/agora/node/QRCodeInterface.d

```diff
@@ -0,0 +1,278 @@
+mixin AddLogger!();
+
+/*******************************************************************************
+
+    Class for Agora User's credentials QR services Interface
+
+    A class to provide a login or encryption key to QRs to verify that
+    the owner of the validator from outside agora.
+
+    This interface listens on 127.0.0.1:2829.
+    Respond SVG format's QR according to the get requests.
+
+    This interface must be in a path accessible only to the owner of the node.
+
+*******************************************************************************/
+
+public class QRCodeInterface
+{
+    /// HTTP listener
+    private HTTPListener listener;
+
+    /// The keypair of this node
+    private KeyPair key_pair;
+
+    /***************************************************************************
+
+        Constructor
+
+        Params:
+            key_pair = the keypair of this node
+
+    ***************************************************************************/
+
+    public this (const KeyPair key_pair)
+    in
+    {
+        assert(KeyPair.init != key_pair);
+    }
+    do
+    {
+        this.key_pair = key_pair;
+    }
+
+    /***************************************************************************
+
+        Start listening for requests
+
+    ***************************************************************************/
+
+    public void start ()
+    {
+        auto settings = new HTTPServerSettings("127.0.0.1");
+        settings.port = 2829;
+        auto router = new URLRouter;
+        router
+            .get("/loginQR", &this.loginQRCreate)
+            .get("/encryptionKeyQR", &this.keyQRCreate);
+        this.listener = listenHTTP(settings, router);
+        scope (exit) this.listener.stopListening();
+    }
+
+    /***************************************************************************
+
+        Respond with QR code containing login information.
+
+        Enter the name of the APP where the login information will be
+        used as the get query.
+
+        Convert LoginInfo to JSON format and respond with the QR code
+        of SVG format.
+        All public key cryptography is done on Curve25519.
+        The signature of the vectorCard provides verification function in the SDK.
+        This LoginInfo has an expiration time.
+        It provide a randomly generated temporary private key encoded
+        in strings on demand.
+        In the future, there is a need to enter a temporary key.
+
+    ***************************************************************************/
+
+    private void loginQRCreate (scope HTTPServerRequest req,
+        scope HTTPServerResponse res)
+    {
+        enforceHTTP("app" in req.query,
+            HTTPStatus.badRequest, "Missing app field.");
+
+        auto app = req.query["app"];
+
+        // The randomly generated temporary KeyPair
+        KeyPair temp_kp = KeyPair.random();
+
+        SysTime expires_time = Clock.currTime();
+        auto expires_term = 1.days;
```

Comment:
1 day is way too short. We should talk to service planning, but for the time being, make it 3 months.

Reply:
I Set this expiration date to 90 days.